src/azure-cli/azure/cli/command_modules/cdn/custom.py

```diff
@@ -492,6 +492,101 @@ def create_custom_domain(client, resource_group_name, profile_name, endpoint_nam
                                         hostname)
 
 
+def enable_custom_https(client, resource_group_name, profile_name, endpoint_name,
+                        custom_domain_name, user_cert_subscription_id=None, user_cert_group_name=None,
+                        user_cert_vault_name=None, user_cert_secret_name=None, user_cert_secret_version=None,
+                        user_cert_protocol_type=None, min_tls_version=None):
+
+    from azure.mgmt.cdn.models import (CdnCertificateSourceParameters,
+                                       UserManagedHttpsParameters,
+                                       CdnManagedHttpsParameters,
+                                       KeyVaultCertificateSourceParameters,
+                                       CertificateType,
+                                       Profile,
+                                       ProtocolType)
+
+    profile: Profile = client.profiles.get(resource_group_name, profile_name)
+
+    if min_tls_version is not None and min_tls_version.casefold() == 'none'.casefold():
+        min_tls_version = MinimumTlsVersion.none
+    elif min_tls_version == '1.0':
+        min_tls_version = MinimumTlsVersion.tls10
+    elif min_tls_version == '1.2':
+        min_tls_version = MinimumTlsVersion.tls12
+
+    # Are we using BYOC?
+    if any(x is not None for x in [user_cert_group_name,
+                                   user_cert_vault_name,
+                                   user_cert_secret_name,
+                                   user_cert_secret_version,
+                                   user_cert_protocol_type]):
+
+        # If any BYOC flags are set, make sure they all are.
+        if any(x is not None for x in [user_cert_subscription_id,
```

Comment:
It looks the condition should be `x is None` to check if there is an element lost, right?

Reply:
Fixed, good catch.